tt_eager/tt_dnn/op_library/reduce/reduce_op.cpp

```diff
@@ -267,11 +268,136 @@ Tensor sum(const Tensor &input_tensor, uint dim, const MemoryConfig& output_mem_
         return AutoFormat::format_output_tensor(output, out_shape, device, Layout::TILE);
     }
 }
+Tensor max(const Tensor &input_tensor, uint dim, const MemoryConfig& output_mem_config) {
```

Comment:
Are there any differences between sum, min, and max besides the reduce enum to pass?

If not can this be refactored into a common function that gets called? Code maintenance on these 3 large functions is non-ideal if it's just duplicated code

Reply:
To clarify I meant to keep the individual max, min, sum api endpoint, but they internally call a common function which takes the enum